trestle/core/commands/author/profile.py

```diff
@@ -536,3 +539,201 @@ def resolve_profile(
+            )
+        except Exception as e:  # pragma: no cover
+            return handle_generic_command_exception(e, logger, 'Profile generation failed')
+
+    @staticmethod
+    def _is_inherited(all_comps: List[ssp.ByComponent]) -> bool:
+        # Fail fast by checking for any non-compliant components.
+        # Must contain provided export statements, no responsibility
+        # statements, and be implemented.
+        for comp in all_comps:
+            if comp.export is None:
+                return False
+
+            if comp.export.responsibilities is not None:
+                return False
+
+            if comp.export.provided is None:
+                return False
+
+            if comp.implementation_status.state != const.STATUS_IMPLEMENTED:
+                return False
+
+        return True
+
+    @staticmethod
+    def update_profile_import(
+        orig_prof_import: prof.Import, leveraged_ssp: ssp.SystemSecurityPlan, catalog_api: CatalogAPI
+    ) -> None:
+        """Add controls to different sections of a profile import based on catalog and leveraged SSP.
+
+        Args:
+            orig_prof_import: The original profile import that will have the control selection updated.
+            leveraged_ssp: SSP input for control filtering
+            catalog_api: Catalog API with access to controls that need to be filtered
+
+        Returns:
+            None
+        """
+        exclude_with_ids: Set[prof.withId] = set()
+        components_by_id: Dict(str, List[ssp.ByComponent]) = {}
+
+        # Create dictionary containing all by-components by control for faster searching
+        for implemented_requirement in leveraged_ssp.control_implementation.implemented_requirements:
+            components: List[ssp.ByComponent] = []
+
+            if implemented_requirement.by_components:
+                components.extend(implemented_requirement.by_components)
+            if implemented_requirement.statements:
+                for stm in implemented_requirement.statements:
+                    if stm.by_components:
+                        components.extend(stm.by_components)
+            components_by_id[implemented_requirement.control_id] = none_if_empty(components)
+
+        # Looping by controls in the catalog because the ids in the profile should
+        # be a subset of the catalog and not the ssp controls.
+        catalog_control_ids: Set[str] = set(catalog_api._catalog_interface.get_control_ids())
+        for control_id in catalog_control_ids:
+
+            if control_id not in components_by_id:
+                continue
+
+            by_comps = components_by_id[control_id]
+            if by_comps is not None and ProfileInherit._is_inherited(by_comps):
+                exclude_with_ids.add(control_id)
+
+        include_with_ids: Set[prof.withId] = catalog_control_ids - exclude_with_ids
+
+        include_controls = [prof.SelectControlById(with_ids=sorted(include_with_ids))]
```

Comment:
where´s include_controls declaration? As we try to keep strongly type variables, please declare variables with its type when possible :)

Reply:
Thanks @AleJo2995! Ended up removing the intermediary variable and just set the value on the import. Please let me know if this adequately addresses your feedback.